platform/admin/core/admin-core-impl/src/main/java/org/codice/ddf/admin/core/impl/AdminAlertImpl.java

```diff
@@ -0,0 +1,120 @@
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.core.impl;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.management.NotCompliantMBeanException;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.shiro.SecurityUtils;
+import org.codice.ddf.admin.core.api.jmx.AdminAlertMBean;
+import org.codice.ddf.persistence.PersistenceException;
+import org.codice.ddf.persistence.PersistentItem;
+import org.codice.ddf.persistence.PersistentStore;
+import org.codice.ddf.system.alerts.Alert;
+import org.codice.ddf.system.alerts.SystemNotice;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.html.HtmlEscapers;
+
+import ddf.security.SubjectUtils;
+
+public class AdminAlertImpl extends BasicMBean implements AdminAlertMBean {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AdminAlertImpl.class);
+
+    private final String activeAlertQuery;
+
+    private final PersistentStore persistentStore;
+
+    private final EventAdmin eventAdmin;
+
+    public AdminAlertImpl(PersistentStore persistentStore, EventAdmin eventAdmin)
+            throws NotCompliantMBeanException {
+        super(AdminAlertMBean.class, AdminAlertMBean.OBJECT_NAME);
+        this.persistentStore = persistentStore;
+        this.eventAdmin = eventAdmin;
+        activeAlertQuery = String.format("status='%s'", Alert.ALERT_ACTIVE_STATUS);
+    }
+
+    @Override
+    public List<Map<String, Object>> getAlerts() {
+        List<Map<String, Object>> alerts = new ArrayList<>();
+        try {
+            List<Map<String, Object>> results = persistentStore.get("alerts", activeAlertQuery);
+            if (!results.isEmpty()) {
+                for (Map<String, Object> item : results) {
+                    item = PersistentItem.stripSuffixes(item);
+                    //this check and type change is needed because of how the persistent store handles
+                    //strings/sets of strings. Single value sets will come back as strings.
+                    if (item.get(SystemNotice.SYSTEM_NOTICE_DETAILS_KEY) instanceof String) {
+                        item.put(SystemNotice.SYSTEM_NOTICE_DETAILS_KEY,
+                                Collections.singleton(item.get(SystemNotice.SYSTEM_NOTICE_DETAILS_KEY)));
+                    }
+                    escapeStrings(item);
+                    alerts.add(item);
+                }
+            }
+        } catch (PersistenceException pe) {
+            LOGGER.debug("Error retrieving system alert.", pe);
+        }
+        return alerts;
+    }
+
+    private void escapeStrings(Map<String, Object> map) {
```

Comment:
✏️ I'm not sure if there's a good reason we use the {{{ }}} braces in the alerts template, but if not, it would be nice to escape the alerts there so input sanitization isn't an issue in the future.